File: Strings/get_longest_palindrome.py
def get_longest_palindrome(string):
    current_longest = [0, 1]
    for i in range(1, len(string)):
        odd = get_palindrome(string, i - 1, i + 1)
        even = get_palindrome(string, i - 1, i)
        longest = max(odd, even, key=lambda x: x[1] - x[0])
        current_longest = max(longest, current_longest, key=lambda x: x[1] - x[0])
    return string[current_longest[0] : current_longest[1]]


def get_palindrome(string, left, right):
    while left >= 0 and right < len(string):
        if string[left] != string[right]:
            break
        left -= 1
        right += 1
    return [left + 1, right]

File: Strings/getLongestPalindrome.js
const longestPalindromicSubstring = (string) => {
    let currentLongest = [0, 1]
    for (let i = 1; i < string.length; i++) {
        const odd = getLongestPalindrome(string, i - 1, i + 1)
        const even = getLongestPalindrome(string, i - 1, i)
        const longest = odd[1] - odd[0] > even[1] - even[0] ? odd : even
        currentLongest = currentLongest[1] - currentLongest[0] > longest[1] - longest[0] ? currentLongest : longest
    }
    return string.slice(currentLongest[0], currentLongest[1])
}

const getLongestPalindrome = (string, left, right) => {
    while (left >= 0 && right < string.length) {
        if (string[left] !== string[right]) break
        left--
        right++
    }
    return [left + 1, right]
}